### PR TITLE
improve test coverage: InMemoryExporter SnapshotMetric

### PR DESCRIPTION
**Pull request by TimothyMothra**
Improving test coverage.
As of today, overall: 85.24%, InMemoryExporter: 78.48%, SnapshotMetric: 60% [source](https://app.codecov.io/gh/open-telemetry/opentelemetry-dotnet)

With this change, SnapshotMetric comes up to 100%

## Changes
- add tests for `SnapshotMetric`.

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by TimothyMothra** [comment]
just did a rebase, trying to fix the codecov report

File: test/OpenTelemetry.Tests/Metrics/MetricSnapshotTests.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            ref readonly var metricPointForFirstExport = ref metricPointsEnumerator.Current;
+            Assert.Equal(10, metricPointForFirstExport.GetSumLong());
+
+            // Verify Snapshot
+            Assert.Single(exportedSnapshots);
+            var snapshot1 = exportedSnapshots[0];
+            Assert.Single(snapshot1.MetricPoints);
+            Assert.Equal(10, snapshot1.MetricPoints[0].GetSumLong());
+
+            // Verify Metric == Snapshot
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Metrics/MetricSnapshotTests.cs:60]: what you really need is to verify that MetricSnapShot is unaffected by subsequent updates, where Metric is affected. - lets add that to the test, if not already done elsewhere.

**Review by cijothomas — APPROVED** [review]
LGTM

File: test/OpenTelemetry.Tests/Metrics/MetricSnapshotTests.cs (hunk 1 above)
Comment (TimothyMothra) [reply test/OpenTelemetry.Tests/Metrics/MetricSnapshotTests.cs:60]: Great idea! I'll get this done today.

**Pull request merged**